x/qgb/abci_test.go

```diff
@@ -83,4 +145,36 @@ func TestValsetSetting(t *testing.T) {
 	require.Equal(t, uint64(1), pk.GetLatestAttestationNonce(ctx))
 }
 
-// Add data commitment window tests
+func TestSetDataCommitment(t *testing.T) {
+	input, ctx := testutil.SetupFiveValChain(t)
+	qk := input.QgbKeeper
+
+	input.Context = ctx.WithBlockHeight(int64(qk.GetDataCommitmentWindowParam(ctx)))
+	vs, err := qk.GetCurrentDataCommitment(ctx)
+	require.Nil(t, err)
+	err = qk.SetAttestationRequest(ctx, &vs)
+	require.Nil(t, err)
+
+	require.Equal(t, uint64(1), qk.GetLatestAttestationNonce(ctx))
+}
+
+func TestDataCommitmentCreation(t *testing.T) {
+	input, ctx := testutil.SetupFiveValChain(t)
+	qk := input.QgbKeeper
+
+	// run abci methods after chain init
+	staking.EndBlocker(input.Context, input.StakingKeeper)
+	qgb.EndBlocker(ctx, *qk)
+
+	// current attestation nonce should be 1 because a valset has been emitted upon chain init.
+	currentAttestationNonce := qk.GetLatestAttestationNonce(ctx)
+	require.Equal(t, uint64(1), currentAttestationNonce)
+
+	// increment height to be the same as the data commitment window
+	newHeight := int64(qk.GetDataCommitmentWindowParam(ctx))
+	input.Context = ctx.WithBlockHeight(newHeight)
+	qgb.EndBlocker(input.Context, *qk)
```

Comment:
We're not interested in block height incremented. All we care about is the `endBlocker` executes, and finds that the `height%DataCommitmentWindow==0` to create a new data commitment

Reply:
> We're not interested in block height incremented

I'm confused. Then why verify this below?

```
	require.Less(t, newHeight, ctx.BlockHeight())
```

Reply:
Sorry, didn't get your message right at first.

This test is setting the height to the data commitment window,  so that  a new data commitment window can be created. Then, executes EndBlocker for the data commitment to be actually created.

Thus, it is testing whether we're at the right height, and a new data commitment was created